src/shell/template_test.go

```diff
@@ -35,3 +35,128 @@ func TestFormatString(t *testing.T) {
 		assert.Equal(t, tc.Expected, got, tc.Case)
 	}
 }
+
+func TestFormatArray(t *testing.T) {
+	text := `{{ formatArray .Value}}`
+	cases := []struct {
+		Case     string
+		Value    interface{}
+		Expected string
+	}{
+		{
+			Case:     "string",
+			Value:    "hello",
+			Expected: `"hello"`,
+		},
+		{
+			Case:     "multipleStrings",
+			Value:    "hello world, I am a long string",
+			Expected: `"hello" "world," "I" "am" "a" "long" "string"`,
+		},
+		{
+			Case:     "bool",
+			Value:    true,
+			Expected: `true`,
+		},
+		{
+			Case:     "int",
+			Value:    32,
+			Expected: `32`,
+		},
+	}
+
+	for _, tc := range cases {
+		got, _ := parse(text, tc)
+		assert.Equal(t, tc.Expected, got, tc.Case)
+	}
+}
+
+func TestEscapeString(t *testing.T) {
+	text := `{{ escapeString .Value}}`
+	cases := []struct {
+		Case     string
+		Value    interface{}
+		Expected string
+	}{
+		{
+			Case:     "string",
+			Value:    `hello`,
+			Expected: `hello`,
+		},
+		{
+			Case:     "stringWithQuotes",
+			Value:    `hello "world"`,
+			Expected: `hello \"world\"`,
+		},
+		{
+			Case:     "stringWithBackslashes",
+			Value:    `hello \world`,
+			Expected: `hello \\world`,
+		},
+		{
+			Case:     "template",
+			Value:    Template(`hello "world"`),
+			Expected: `hello \"world\"`,
+		},
+	}
+
+	for _, tc := range cases {
+		got, _ := parse(text, tc)
+		assert.Equal(t, tc.Expected, got, tc.Case)
+	}
+}
+
+func TestMatch(t *testing.T) {
+	text := `{{ match .Variable "hello" "world"}}`
+	cases := []struct {
+		Case     string
+		Variable string
+		Expected string
+	}{
+		{
+			Case:     "match",
+			Variable: "hello",
+			Expected: `true`,
+		},
+		{
+			Case:     "match",
+			Variable: "world",
+			Expected: `true`,
+		},
+		{
+			Case:     "noMatch",
+			Variable: "goodbye",
+			Expected: `false`,
+		},
+	}
+
+	for _, tc := range cases {
+		got, _ := parse(text, tc)
+		assert.Equal(t, tc.Expected, got, tc.Case)
+	}
+}
+
+func TestHasCommand(t *testing.T) {
+	text := `{{ hasCommand .Command}}`
+	cases := []struct {
+		Case     string
+		Command  string
+		Expected string
+	}{
+		{
+			Case:     "hasCommand",
+			Command:  "ls",
```

Comment:
this should fail on Windows

Reply:
This actually passes on my windows parallels environment in powershell:
```
PS Y:\aliae\src\shell> go test
PASS
ok      github.com/jandedobbeleer/aliae/src/shell       0.274s
```

`ls` is a default alias for `Get-ChildItem` so it should be available on all PS installs 6 and up

Reply:
Annnnnd it doesn't work on my windows PC, so it's some parallels thing doing it.
Back to the drawing board!

Reply:
Duh, everyone who is running the test has to have `go` in their path!